### PR TITLE
Adds pandas Methods (Part I)

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
Methods: 'abs', 'add', 'add_prefix', 'add_suffix', 'agg', 'aggregate', 'align', 'all', 'any', 'append', 'apply', 'applymap'.

#### Related Issue
Ref.: #4284 

#### Motivation and Context
Improve index support for pandas.DataFrame for history request.

#### How Has This Been Tested?
New unit tests.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Python/PandasData.cs [hunk 1]
```diff
@@ -133,33 +135,89 @@ class Remapper(wrapt.ObjectProxy):
...
+        result = self.__wrapped__.any(axis=axis, bool_only=bool_only, skipna=skipna, level=level, **kwargs)
         return Remapper(result)
 
     def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
-        result = self.__wrapped__.append(other=other, ignore_index=ignore_index, verify_integrity=verify_integrity, sort=sort)
+        '''https://pandas.pydata.org/pandas-docs/version/0.25.1/reference/api/pandas.DataFrame.append.html'''
+        result = self.__wrapped__.append(other, ignore_index=ignore_index, verify_integrity=verify_integrity, sort=sort)
+        return Remapper(result)
+
+    def apply(self, func, axis=0, broadcast=False, raw=False, reduce=None, args=(), **kwds):
```
Comment (Martin-Molinero) [inline Common/Python/PandasData.cs:188]: Missing `result_type=None,` and `broadcast` default is `None`?

File: Common/Python/PandasData.cs [hunk 2]
```diff
@@ -104,20 +104,22 @@ class Remapper(wrapt.ObjectProxy):
 
         return key
 
+    def _wrap_if_pandas_object(self, result):
+        ''' For these cases we wrap the result too. Can't apply the wrap around all
+        results because it causes issues in pandas for some of our use cases
+        specifically pandas timestamp type'''
+        if isinstance(result, (pandas.Series, pandas.DataFrame, pandas.Index)):
```
Comment (gsalaz98) [inline Common/Python/PandasData.cs:111]: Do we want to support `MultiIndex` objects in the future?

File: Tests/Python/PandasConverterTests.cs [hunk 3]
```diff
@@ -181,6 +181,201 @@ public void HandlesQuoteBars()
...
+            if (cache) SymbolCache.Set("SPY", Symbols.SPY);
+
+            using (Py.GIL())
+            {
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+                    $@"
+def Test(dataFrame, symbol):
+    data = dataFrame.agg('mean', axis=0)
+    className=str(type(data))
+    if className.find('Remapper') < 0:
```
Comment (Martin-Molinero) [inline Tests/Python/PandasConverterTests.cs:284]: Could we adjust `_agg` and `_aggregate` to test using the index too?

Comment (AlexCatarino) [reply Tests/Python/PandasConverterTests.cs:284]: I tried every combination of aggregation of index and column (`axis=0` and `axis=1`) and it never got a `Series` or `DataFrame` with the index, so I asserted the type.

**Pull request merged**